### PR TITLE
Add missing param for parent call.

### PR DESCRIPTION
**Pull request by dereuromark**
right?

### CODE REVIEW

File: src/ORM/Query.php [hunk 1]
```diff
@@ -1229,7 +1229,7 @@ public function delete(?string $table = null)
         $repository = $this->getRepository();
         $this->from([$repository->getAlias() => $repository->getTable()]);
 
-        return parent::delete();
+        return parent::delete($table);
```
Comment (markstory) [inline src/ORM/Query.php]: This will overwrite the table + alias set on line 1230 which I don't think is ideal.

**Comment by saeideng** [comment]
you can remove `?string $table = null` parameter
it never used in 3.x series https://github.com/cakephp/cakephp/commit/0d6f20251fb99fd6efa6d624cb24a73e2a8faca8

File: src/ORM/Query.php (hunk 1 above)
Comment (dereuromark) [reply src/ORM/Query.php]: Then we should add a comment why the parent is called without the argument, as this is not usual :)

**Comment by markstory** [comment]
@saeideng But someone else could be using it.

**Comment by dereuromark** [comment]
Closing in favor of https://github.com/cakephp/cakephp/pull/12877 then

**Comment by dereuromark** [comment]
@markstory We cannot clean this up for 4.x?

**Pull request closed without merging**